packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +921,977 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+
+    this.assertText('Foo4Bar5');
+
+    this.runTask(() => this.context.get('things').shiftObject());
+
+    this.assertText('4Bar5');
+
+    this.runTask(() => this.context.get('things').clear());
+
+    this.assertText('');
+
+    this.runTask(() => this.context.set('things', emberA(['Foo', 4, 'Bar'])));
+
+    this.assertText('Foo4Bar');
+  }
+
+  ['@htmlbars can use hash parameter instead of positional param'](assert) {
+    this.registerComponent('sample-component', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['first', 'second']
+      }),
+      template: '{{first}} - {{second}}'
+    });
+
+    this.render(strip`
+      {{sample-component "one" "two" id="two-positional"}}
+      {{sample-component "one" second="two" id="one-positional"}}
+      {{sample-component first="one" second="two" id="no-positional"}}`);
+
+    assert.equal(this.$('#two-positional').text(), 'one - two');
+    assert.equal(this.$('#one-positional').text(), 'one - two');
+    assert.equal(this.$('#no-positional').text(), 'one - two');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#two-positional').text(), 'one - two');
+    assert.equal(this.$('#one-positional').text(), 'one - two');
+    assert.equal(this.$('#no-positional').text(), 'one - two');
+  }
+
+  ['@skip dynamic arbitrary number of positional parameters'](assert) {
+    this.registerComponent('sample-component', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: 'n'
+      }),
+      template: strip`
+        {{#each n as |name|}}
+          {{name}}
+        {{/each}}`
+    });
+
+    this.render(strip`
+      {{sample-component user1 user2 id="direct"}}
+      {{component "sample-component" user1 user2 id="helper"}}`,
+      {
+        user1: 'Foo',
+        user2: 4
+      }
+    );
+
+    assert.equal(this.$('#direct').text(), 'Foo4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Foo4', 'helper');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#direct').text(), 'Foo4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Foo4', 'helper');
+
+    this.runTask(() => this.context.set('user1', 'Bar'));
+
+    assert.equal(this.$('#direct').text(), 'Bar4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Bar4', 'helper');
+
+    this.runTask(() => this.context.set('user2', '5'));
+
+    assert.equal(this.$('#direct').text(), 'Bar5', 'direct');
+    assert.equal(this.$('#helper').text(), 'Bar5', 'helper');
+
+    this.runTask(() => {
+      this.context.set('user1', 'Foo');
+      this.context.set('user2', 4);
+    });
+
+    assert.equal(this.$('#direct').text(), 'Foo4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Foo4', 'helper');
+  }
+
+  ['@htmlbars with ariaRole specified']() {
+    this.registerComponent('aria-test', {
+      template: 'Here!'
+    });
+
+    this.render('{{aria-test id="aria-test" ariaRole="main"}}');
```

Comment:
1. I think we don't need the ID anymore (it was only there before so we can jQuery it)
2. make ariaRole dynamic for I-N-U-R (see https://github.com/emberjs/ember.js/pull/12890/files#r58618076)